src/ApiService/ApiService/Functions/AgentCanSchedule.cs

```diff
@@ -40,7 +40,12 @@ public class AgentCanSchedule {
         var allowed = canProcessNewWork.IsAllowed;
         var reason = canProcessNewWork.Reason;
 
-        var task = await _context.TaskOperations.GetByTaskId(canScheduleRequest.TaskId);
+        var task = await
+            (canScheduleRequest.JobId.HasValue
+            ? _context.TaskOperations.GetByJobIdAndTaskId(canScheduleRequest.JobId.Value, canScheduleRequest.TaskId)
+            // old agent, fall back
```

Comment:
We should also create an issue to remove the old agent case in the release after this one (9.1 I guess?)